public/app/features/dashboard-scene/settings/VariablesEditView.tsx

```diff
@@ -19,11 +19,93 @@ export class VariablesEditView extends SceneObjectBase<VariablesEditViewState> i
   static Component = ({ model }: SceneComponentProps<VariablesEditView>) => {
     const dashboard = getDashboardSceneFor(model);
     const { navModel, pageNav } = useDashboardEditPageNav(dashboard, model.getUrlKey());
+    // get variables from dashboard state
+    const variablesObject = sceneGraph.getVariables(dashboard);
+    const variables = variablesObject.useState().variables;
+
+    const getVariableIndex = (identifier: string) => {
+      return variables.findIndex((variable) => variable.state.name === identifier);
+    };
+
+    const onDelete = (identifier: string) => {
+      // Find the index of the variable to be deleted
+      const variableIndex = getVariableIndex(identifier);
+      if (variableIndex === -1) {
+        // Handle the case where the variable is not found
+        console.error('Variable not found');
+        return;
+      }
+
+      // Create a new array excluding the variable to be deleted
+      const updatedVariables = [...variables.slice(0, variableIndex), ...variables.slice(variableIndex + 1)];
+
+      // Update the state or the variables array
+      variablesObject.setState({ variables: updatedVariables });
+    };
+
+    const onDuplicated = (identifier: string) => {
+      const variableIndex = getVariableIndex(identifier);
```

Comment:
@axelavargas - those handlers are better suited to the model. This way you separate the rendering from the operational logic nicely, and your component is literally a few-liner then :)

Reply:
And you can easily test the model without the need for rendering it :)

Reply:
I'm referring to the model of this view, which is `VariablesEditView`. You can also implement not inside the model, but as a regular component (see https://grafana.com/developers/scenes/core-concepts#scene-object).

Reply:
@dprokop I have made the refactor, also added unit test :),